presto-main/src/main/java/com/facebook/presto/execution/CreateTableTask.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.metadata.QualifiedTableName;
+import com.facebook.presto.metadata.TableHandle;
+import com.facebook.presto.metadata.TableMetadata;
+import com.facebook.presto.spi.ColumnMetadata;
+import com.facebook.presto.spi.ConnectorTableMetadata;
+import com.facebook.presto.sql.analyzer.SemanticException;
+import com.facebook.presto.sql.tree.CreateTable;
+import com.facebook.presto.sql.tree.TableElement;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeSignature;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.metadata.MetadataUtil.createQualifiedTableName;
+import static com.facebook.presto.sql.analyzer.SemanticErrorCode.TABLE_ALREADY_EXISTS;
+import static com.facebook.presto.sql.analyzer.SemanticErrorCode.MISSING_COLUMN_DEFINITION;
+
+public class CreateTableTask
+        implements DataDefinitionTask<CreateTable>
+{
+    @Override
+    public String getName()
+    {
+        return "CREATE TABLE";
+    }
+
+    @Override
+    public void execute(CreateTable statement, Session session, Metadata metadata, QueryStateMachine stateMachine)
+    {
+        QualifiedTableName tableName = createQualifiedTableName(session, statement.getTable());
+        Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
+        if (tableHandle.isPresent()) {
+            throw new SemanticException(TABLE_ALREADY_EXISTS, statement, "Table '%s' already exists", tableName);
+        }
+
+        if (statement.getTableElement().isEmpty()) {
```

Comment:
It should not be possible for this to happen since the grammar requires at least one item, thus we don't need a user error code (it would be a bug in Presto):
```java
checkArgument(!statement.getTableElement().isEmpty(), "no columns for table");